lib/linter-slim.js

```diff
@@ -0,0 +1,136 @@
+'use babel';
+
+// eslint-disable-next-line import/no-extraneous-dependencies, import/extensions
+import { CompositeDisposable } from 'atom';
+
+let helpers;
+
+// Deprecated syntax:
+const REGEX = /.*(SyntaxError|Deprecated).*: (.+)\n.+, Line (\d+), Column (\d+)/g;
+
+const severityMapping = {
+  deprecated: 'warning',
+  syntaxerror: 'error',
+};
+
+const loadDeps = () => {
+  if (!helpers) {
+    helpers = require('atom-linter');
+  }
+};
+
+const parseCompileOutput = (output, filePath, editor) => {
+  const messages = [];
+
+  let match = REGEX.exec(output);
+  while (match !== null) {
+    const severity = severityMapping[match[1].toLowerCase()] || 'warning';
+    const line = Number.parseInt(match[3], 10) - 1;
+    const col = Number.parseInt(match[4], 10) - 1;
+    const excerpt = `${match[1]}: ${match[2]}`;
+    messages.push({
+      severity,
+      excerpt,
+      location: {
+        file: filePath,
+        position: helpers.generateRange(editor, line, col),
+      },
+    });
+    match = REGEX.exec(output);
+  }
+  return messages;
+};
+
+module.exports = {
+  activate() {
+    this.idleCallbacks = new Set();
+    let depsCallbackID;
+    const installLinterReekDeps = () => {
+      this.idleCallbacks.delete(depsCallbackID);
+      if (!atom.inSpecMode()) {
+        require('atom-package-deps').install('linter-slim');
+      }
+      loadDeps();
+    };
+    depsCallbackID = window.requestIdleCallback(installLinterReekDeps);
+    this.idleCallbacks.add(depsCallbackID);
+
+    this.subscriptions = new CompositeDisposable();
+    this.subscriptions.add(
+      atom.config.observe(
+        'linter-slim.slimrbExecutablePath',
+        (value) => { this.slimrbExecutablePath = value; },
+      ),
+      atom.config.observe(
+        'linter-slim.rails',
+        (value) => { this.rails = value; },
+      ),
+      atom.config.observe(
+        'linter-slim.library',
+        (value) => { this.library = value; },
+      ),
+    );
+  },
+
+  deactivate() {
+    this.idleCallbacks.forEach(callbackID => window.cancelIdleCallback(callbackID));
+    this.idleCallbacks.clear();
+    this.subscriptions.dispose();
+  },
+
+  provideLinter() {
+    return {
+      name: 'slimrb',
+      grammarScopes: ['text.slim'],
+      scope: 'file',
+      lintsOnChange: true,
+      lint: async (editor) => {
+        loadDeps();
+
+        const filePath = editor.getPath();
+        if (!filePath) {
+          return null;
+        }
+
+        const args = [];
+        args.push('--compile');
+
+        if (this.rails) {
+          args.push('--rails');
+        }
+
+        if (Array.isArray(this.library) && this.library.length) {
```

Comment:
Atom's configuration schema ensures that the result of `atom.config.observe` will always be an `Array`.
```suggestion
        if (this.library.length > 0) {
```